token-lending/program/src/processor.rs

```diff
@@ -778,59 +816,28 @@ fn process_borrow(
         return Err(LendingError::InvalidMarketAuthority.into());
     }
 
-    // deposit collateral
+    obligation
+        .find_or_add_collateral(*deposit_reserve_info.key)?
+        .deposit(collateral_amount)?;
+    // @TODO: instead of always marking stale, we could look for an optional
```

Comment:
If you always mark stale, you could still have a "refresh obligation" instruction after this before processing the borrow right?

Reply:
You could, and then as a client you never need to know about the `aggregator_info` that would be needed except when refreshing reserves. I planned to test this out and see if it's feasible to fit everything (refresh reserve, deposit to obligation, refresh obligation, borrow liquidity) in a single trx, so I think this may work.